client/my-sites/theme/main.jsx

```diff
@@ -885,20 +885,31 @@ class ThemeSheet extends Component {
 		}
 
 		if ( hasWpOrgThemeUpsellBanner || hasThemeUpsellBannerAtomic ) {
+			const thisPageUrl = `/theme/${ themeId }/${ siteSlug }`;
 			pageUpsellBanner = (
 				<UpsellNudge
 					plan={ PLAN_BUSINESS }
 					className="theme__page-upsell-banner"
-					title={ translate( 'Access this theme for FREE with a Business plan!' ) }
+					title={
+						siteId
+							? translate( 'Upgrade your plan to access third-party themes' )
+							: translate( 'Access third-party themes with a Business plan' )
+					}
 					description={ preventWidows(
 						translate(
-							'Instantly unlock thousands of different themes and install your own when you upgrade.'
+							'Instantly unlock thousands of different themes and install your own when you upgrade to the Business plan.'
 						)
 					) }
 					forceHref
 					feature={ FEATURE_UPLOAD_THEMES }
 					forceDisplay
-					href={ ! siteId ? '/plans' : null }
+					href={
+						siteId
+							? `/checkout/${ siteSlug }/business?redirect_to=${ encodeURIComponent(
+									thisPageUrl
+							  ) }`
```

Comment:
Nitpick: `encodeURIComponent` replaces `/` with `%2F` (e.g. `?redirect_to=%2Ftheme%2Fastra%2Fexample.wordpress.com#step2`), but slashes don't need to be escaped (`?redirect_to=/theme/astra/example.wordpress.com#step2` works too). Since `thisPageUrl` is defined a few lines above (i.e. it is not given as param), we can guarantee it doesn't include any character that needs to be escaped and simplify this a little bit.
```suggestion
							? `/checkout/${ siteSlug }/business?redirect_to=${ thisPageUrl }`
```

Reply:
Cheers!